src/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -17,5 +21,136 @@ private Socket GetOrCreateAcceptSocket(Socket acceptSocket, bool unused, string
             handle = null;
             return null;
         }
+
+        private void SendFileInternal(string fileName, byte[] preBuffer, byte[] postBuffer, TransmitFileOptions flags)
+        {
+            // Note, UseDefaultWorkerThread is the default and is == 0.
+            // Unfortunately there is no TransmitFileOptions.None.
+            if (flags != TransmitFileOptions.UseDefaultWorkerThread)
+            {
+                throw new PlatformNotSupportedException();
+            }
+
+            // Open the file, if any
+            // Open it before we send the preBuffer so that any exception happens first
+            FileStream fileStream = null;
+            if (fileName != null && fileName.Length > 0)
+            {
+                fileStream = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.Read);
+            }
+
+            SocketError errorCode = SocketError.Success;
+            using (fileStream)
+            {
+                // Send the preBuffer, if any
+                // This will throw on error
+                if (preBuffer != null && preBuffer.Length > 0)
+                {
+                    Send(preBuffer);
+                }
```

Comment:
So that we close the fileStream in case of an exception from Send.  

See also the comment above (line 35): we open the file first in case there's an error opening the file.  

I made it work this way because it seemed like the best option -- matches Windows behavior, and it's probably what the user would expect.